hal/common/critical.c

```diff
@@ -270,7 +270,7 @@ uint32_t core_util_atomic_incr_u32(uint32_t *valuePtr, uint32_t delta)
 }
 
 void *core_util_atomic_incr_ptr(void **valuePtr, ptrdiff_t delta) {
-    return core_util_atomic_incr((uintptr_t)valuePtr, (uintptr_t)delta);
+    return core_util_atomic_incr_u32((uintptr_t)valuePtr, (uintptr_t)delta);
```

Comment:
> Let me know if that is still confusing.

I am missing the example how this API shall be used, that might explain my question about why return is ``void*``